### DISCUSSION TITLE
Messages Requests

### DISCUSSION THREAD

**Discussion by michaelgiordanod3** [deso-protocol/dips#21] [Medium CIPs]
Two items up for discussion.

- Upgraded Block Feature. Block should extend not only to the Feed, but also to Messages. If a user chooses to block someone from their feed, it is safe to assume they don't want Messages from that person either.

- Messages should have a delete/hide feature. The filter is nice but being able to delete the unsolicited Messages from Bots/Scam accounts is nice UI/UX feature. We want to show that we are a community of creators, checking messages and seeing those types of messages is not ideal. I like to see messages from those I do not follow yet, but dislike the Bot Messages / SPAM & Scam messages that I also see once turn filters off.

**Comment by HPaulson** [comment]
Both seem to be easily implementable backend changes -- however I should note, in order to build 2 without overcomplicating it, you'd need such to be on a node-by-node (or device-by-device) basis. If you have this in the ideal scenario, where you'd be able to hide conversations on all devices, you'd need to create a new transaction type -- which seems excessive for such a change.

The first is definitely a +1 for me, the second seems like a great idea but the implementation could be complex in the ideal scenario, so it's likely favorable to take a simpler but less ideal approach (Such as having the node itself store such data, like it used to do with blocking)

**Reply by michaelgiordanod3** [reply]
So striking item two from the list, what would next steps be for item 1.

For item 2, would adding a filter to remove blocked users (to the four that already exist) be the simpler option to implement?

**Reply by HPaulson** [reply]
@michaelgiordanod3 Well, if it's just a filter for blocked users, then it's actually the same as 1 -- which is fine. 

Since this is only a medium change, it will only require one core dev approval to move on to the next step. Having some additional community input/approval would also be nice here though, c.c @tijno @dgsus @mubashariqbal

**Comment by tijno** [comment]
I agree with not showing DMs from users i have blocked. 

Agree with above discussion conclusion that the 2 original suggestions are probably covered by this one change.

One note - blocks are due to move on chain soon.

Im not sure if this would impact things but worth considering.

**Reply by HPaulson** [reply]
@tijno I think this change would have to wait until they're on-chain -- kinda defeats the purpose otherwise from my view (As it'd be node-by-node, which is annoying)

**Reply by tijno** [reply]
Im fine either way - before or after. 

Its not a massive issue for me - more a nice to have. 

Cant wait for blocks to be on chain :)
